cat.plugin.ts

```diff
@@ -0,0 +1,85 @@
+import Constants from '../../common/constants';
+import { Plugin } from '../../common/plugin';
+import { ChannelType, IContainer, IHttpResponse, IMessage } from '../../common/types';
+
+class Breed {
+    public name: string = '';
+    public id: string = '';
+}
+
+export class CatPlugin extends Plugin {
+    public name: string = 'Cat Plugin';
+    public description: string = 'Generates pictures of cats.';
+    public usage: string = 'cat <breed (optional)>';
+    public permission: ChannelType = ChannelType.Public;
+    public pluginChannelName: string = Constants.Channels.Public.Pets;
+
+    private _API_URL: string = 'https://api.thecatapi.com/v1/';
+    private _breeds: Breed[] = [];
+
+    constructor(public container: IContainer) {
+        super();
+        //creates list of breeds
+        this.container.httpService
+            .get(`${this._API_URL}breeds`)
+            .then((response: IHttpResponse) => {
+                const breeds = response.data;
+
+                this._breeds = breeds.map((breedData: { name: string, id: string }) => {
+                    return {
+                        name: breedData.name.toLowerCase(),
+                        id: breedData.id.toLowerCase(),
+                    }
+                })
+            })
+            .catch(console.log);
+    }
+
+    public async execute(message: IMessage, args?: string[]) {
+        if (args != undefined && (args[0] === "breeds" || args[0] === "breed")) {
+            //Simply return the list of suported breeds
+            let reply = "";
+
+            this._breeds.forEach(breed => {
```

Comment:
this can just be `const reply = this._breeds.join('\n');`

Reply:
not sure why reply would be "this._breeds"  as that is the list of breeds that are available, I believe you meant "breed.name" which is the breed that the user requested.  However, unless I get rid of the random feature this is the neatest way to do it.  If their is a reason that you do not want  me to use let I can change it.